drivers/dma/dma_intel_adsp_hda_host_in.c

```diff
@@ -19,12 +19,47 @@ static const struct dma_driver_api intel_adsp_hda_dma_host_in_api = {
 	.chan_filter = intel_adsp_hda_dma_chan_filter,
 };
 
+#if CONFIG_DMA_INTEL_ADSP_HDA_TIMING_L1_EXIT
+#include <adsp_interrupt.h>
 #define INTEL_ADSP_HDA_DMA_HOST_IN_INIT(inst)                                                      \
```

Comment:
INTEL_ADSP_HDA_DMA_HOST_*_INIT defined two times based on kconfig, looks ugly, any smart DT idea to merge them?

Reply:
You can use some macros to only initialize .irq_config if L1_EXIT is defined, e.g.

IF_ENABLED(CONFIG_DMA_DMA_INTEL_ADSP_HDA_TIMING_L1_EXIT, (.irq_config = ...,)) \

Could at least re-use the same INIT macro then hopefully avoiding some duplication

Reply:
Thanks, works perfectly, ended with IF_ENABLED in irq_config and COND_CODE_1 in config init